src/main/java/appeng/me/cache/EnergyGridCache.java

```diff
@@ -643,4 +678,75 @@ public boolean unregisterEnergyInterest( final EnergyThreshold threshold )
 		return this.interests.remove( threshold );
 	}
 
+	private class GridPowerStorage implements IAEPowerStorage
+	{
+		double stored = 0;
+
+		@Override
+		public double extractAEPower( double amt, Actionable mode, PowerMultiplier usePowerMultiplier )
+		{
+			double extracted = Math.min( amt, this.stored );
+
+			if( mode == Actionable.MODULATE )
+				this.stored -= extracted;
+
+			if( this.stored < MAX_BUFFER_STORAGE - 0.001 )
+			{
+				EnergyGridCache.this.myGrid.postEvent( new MENetworkPowerStorage( this, PowerEventType.REQUEST_POWER ) );
+			}
+
+			if( extracted < amt )
+			{
+				extracted += EnergyGridCache.this.extractAEPower( amt - extracted, Actionable.MODULATE, PowerMultiplier.CONFIG );
+			}
+
+			return extracted;
+		}
+
+		@Override
+		public boolean isAEPublicPowerStorage()
+		{
+			return true;
+		}
+
+		@Override
+		public double injectAEPower( double amt, Actionable mode )
+		{
+			double toStore = Math.min( amt, MAX_BUFFER_STORAGE - this.stored );
+
+			if( mode == Actionable.MODULATE )
+				this.stored += toStore;
+
+			return amt - toStore;
+		}
+
+		@Override
+		public AccessRestriction getPowerFlow()
+		{
+			return AccessRestriction.READ_WRITE;
+		}
+
+		@Override
+		public double getAEMaxPower()
+		{
+			return MAX_BUFFER_STORAGE;
+		}
+
+		@Override
+		public double getAECurrentPower()
+		{
+			return this.stored;
+		}
+
+		public void setAECurrentPower( double amount )
+		{
+			this.stored = amount;
+		}
+
+		public void addAECurrentPower( double amount )
+		{
+			this.stored += amount;
+		}
+	};
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)